### ISSUE TITLE
Config-driven experiment runner for reproducible multi-seed benchmarks

### ISSUE THREAD

**Issue by hyang0129** [hyang0129/HalluLens#9]
## Summary

Replace notebook-driven experiment workflow with a config-driven CLI that runs all detection methods across datasets with multi-seed reproducibility.

Design plan: `docs/planning/EXPERIMENT_RUNNER_PLAN.md`

## What this enables

- Run all 5 methods (contrastive, logprob-recon, linear probe, token entropy, logprob baseline) from a single command
- Multi-seed sweeps (5 training seeds, fixed split seed) for publishable confidence intervals
- Structured artifact storage: per-run `config.json`, `eval_metrics.json`, `predictions.csv`, `train_metrics.jsonl`
- Aggregation script that produces mean ± 95% CI tables across seeds and datasets

## Key design decisions

- **Three-layer config**: dataset configs, method configs, experiment configs (composed at runtime)
- **Two split strategies**: external test (HotpotQA) vs internal 3-way split (NQ) — see below
- **Two-level seeding**: fixed `split_seed=42` for data splits, varied `training_seeds=[0,5,26,42,63]` for weight init/batch order
- **Non-learned methods** (token entropy, logprob) run once per dataset (deterministic)
- **Resume-safe**: skips completed (method, seed) combos if `eval_metrics.json` exists

### Strategy A: External test set (HotpotQA)

Upstream benchmark provides separate train/test splits. Both have activations.

```
hotpotqa_train (77,673 activations)          hotpotqa (7,405 activations)
├── train (85%)  ── split_seed ──►           │
├── val   (15%)  ── split_seed ──►           │
                                             └── test (100%, no splitting)
```

- `split_seed` controls train/val boundary only
- Test set is always the full external set — identical for every method and seed

### Strategy B: Internal 3-way split (Natural Questions)

Single dataset, no external split. Runner does stratified 3-way split.

```
natural_questions_logprob (20,772 activations)
├── train (70%)  ── split_seed ──►
├── val   (15%)  ── split_seed ──►
└── test  (15%)  ── split_seed ──►
```

- `split_seed` controls all boundaries via two sequential `train_test_split` calls
- Same seed = same train/val/test for every method

**Both strategies guarantee**: for a given `split_seed`, every method sees the exact same data in each split.

## Implementation tasks

- [ ] Create `configs/` directory structure with dataset, method, and experiment JSON configs
- [ ] Modify `ActivationParser` to support 3-way splits (train/val/test) — additive change, 2-way preserved
- [ ] Implement Strategy A loader (two `ActivationParser` instances, train split further into train+val)
- [ ] Implement Strategy B loader (one `ActivationParser`, internal 3-way split)
- [ ] Extract eval JSON builder from notebook into `utils/eval_builder.py`
- [ ] Extract logprob baseline scorer into `activation_research/baselines.py`
- [ ] Implement `seed_everything()` + `worker_init_fn` (currently training is mostly unseeded)
- [ ] Implement `scripts/run_experiment.py` — main runner with (dataset × method × seed) loop
- [ ] Implement `scripts/aggregate_results.py` — collect results, compute mean ± CI, output tables
- [ ] Add `run_manifest.json` generation (git hash, environment, timestamp)
- [ ] Validate: reproduce notebook results for hotpotqa contrastive + linear probe
- [ ] Multi-seed validation: 5-seed sweep on hotpotqa, verify CI ranges

### HotpotQA — Strategy A (external test set)
| Directory | HF split | Activations | Generations | Role |
|---|---|---|---|---|
| `shared/hotpotqa_train` | `train` (90K) | 77,673 | 78,116 (`output/hotpotqa_train/`) | Train → split into train+val |
| `shared/hotpotqa` | `validation` (7,405) | 7,405 | 7,405 (`output/hotpotqa/`) | Test (used as-is) |

### Natural Questions — Strategy B (internal 3-way split)
| Directory | Source | Activations | Generations | Role |
|---|---|---|---|---|
| `shared/natural_questions_logprob` | `nq_wc_dataset.csv` (20,772) | 20,772 | 20,772 (`shared/.../natural_questions/`) | All → split into train/val/test |

### Other datasets (older format, may need migration)
| Dataset | Activations | Generations | Notes |
|---|---|---|---|
| `shared/goodwiki_jsonv2` | JSON | 59,049 | PreciseWikiQA, `h_score_cat` labels |
| `shared/triviiqa_json` | JSON | 20,000 | TriviaQA, exact match labels |
| `shared/precisewikilogprob` | 28,067 (Zarr) | TBD | PreciseWikiQA with logprobs |

## Related

- `PAPER_ROADMAP.md` — overall paper strategy
- `docs/reference/RESULTS_SCHEMA.md` — artifact spec

**Comment by hyang0129** [comment]
## Pre-implementation check

Plan produced: `ISSUE_9_PLAN.md`
Tier: 2 (parallel Coders + sequential Integrator)

Open questions resolved from issue description:
- Val split: included (issue explicitly describes 3-way splits)
- Strategy A + B: both in scope (issue describes both)
- logprob_recon: included (listed in 5 methods)
- Config format: JSON (simplest)
- Dataset paths: relative to repo root
- W&B: deferred to follow-up

No blocking decisions. Proceeding to implementation.
